etc/Jenkinsfile.local

```
@@ -0,0 +1,38 @@
+node {
+    def mvnHome = tool 'mvn-default'
+
+    stage ('Checkout') {
+        checkout scm
+    }
+
+    stage ('Build and Static Analysis') {
+        withMaven(maven: 'mvn-default', mavenLocalRepo: '/var/data/m2repository', mavenOpts: '-Xmx768m -Xms512m') {
```

Shouldn't the maven local repository be something like '~/.m2/repository'?

---

Sorry, this is my local setup from https://github.com/uhafner/warnings-ng-plugin-devenv. It automatically builds on an agent. You can adapt that to fit your needs. I just wanted to give you an example...

---

I think it makes sense we use the default since a lot of people are using the default. And for custom users, the default can be overridden by `<localRepository>` in ~/.m2/settings.xml.

```suggestion
        withMaven(maven: 'mvn-default', mavenOpts: '-Xmx768m -Xms512m') {
```